### ISSUE TITLE
Saving Gray8 image as jpeg results in 24 bit image

### ISSUE THREAD

**Issue by CoenraadS** [SixLabors/ImageSharp#807]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of ImageSharp
- [x] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [x] I have searched [open](https://github.com/SixLabors/ImageSharp/issues) and [closed](https://github.com/SixLabors/ImageSharp/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

I found this: 
https://github.com/SixLabors/ImageSharp/issues/672#issuecomment-411172998

>@jherby2k in our current design Image<T> does not preserve the bit rate / color model of the original encoded data.

And this:

https://stackoverflow.com/questions/52407878/load-and-save-opaque-8-bit-png-files-using-imagesharp
>Note. The png encoder by default will save the image in the input color type and bit depth. If you want to encode the image in a different color type you will need to new up an PngEncoder instance with the ColorType and BitDepth properties set.

### Description
I am loading/resize an 8 bit png, however I want to save as .jpg

JpegEncoder does not have `ColorType` or `BitDepth`

### Steps to Reproduce
Input Image:
![201812211019388645_validpos_19 19875_21 285_439_12](https://user-images.githubusercontent.com/5593825/51072974-8b3ebd80-166a-11e9-95c6-c97ec0a542ac.png)

```
private static void resizeUsingImageSharp(byte[] bytes, MemoryStream output)
        {
            var encoder = new JpegEncoder();
            encoder.Quality = 10;

            var decoder = new PngDecoder();
            decoder.IgnoreMetadata = true;

            var image = Image.Load<Gray8>(bytes, decoder);

            image.Mutate(ctx => ctx.Resize(96, 96));
            // Using .Grayscale() makes no difference, still saves 24 bit image

            // encoder.Encode<Gray8>(image, output); also doesn't work

            image.SaveAsJpeg(output, encoder);
        }

```

Output Image:
![imagesharp](https://user-images.githubusercontent.com/5593825/51073012-2768c480-166b-11e9-812f-0cfc96e2af5f.jpg)

### System Configuration

- ImageSharp version:
```
<PackageReference Include="SixLabors.ImageSharp" Version="1.0.0-dev002321" />
<PackageReference Include="SkiaSharp" Version="1.68.0" />
```

- Environment (Operating system, version and so on):
Window 10
- .NET Framework version:
`netcoreapp3.0`
- Additional information:

**Comment by JimBobSquarePants** [comment]
@CoenraadS Jpegs are 24 bit (8 bits per channel). You are never going to be able to save an 8bit jpeg.

The difference in size is probably due to a combination of different output quality, metadata preservation and resampling algorithm. 

The performance difference in Resize is dubious, like-for-like benchmarks show that [we normally come out on top and also produce higher quality output](https://github.com/JimBobSquarePants/core-imaging-playground). This does depend on many factors, CPU etc.  If you're saving at 10% quality you should be using Nearest Neighbor for resizing as it's way faster.

SkiaSharp decodes and encode jpegs faster because they use libjpeg-turbo under the hood. We do not have the SIMD tools available yet to allow faster decoding but we can probably speed up encoding a lot by optimizing our color transforms (It's on the TODO list).

Once you align the output quality you'll find the performance much closer. 

Anyway it's milliseconds and our API is an order of magnitude nice to use.

**Comment by CoenraadS** [comment]
Aw good to know. I was also comparing with XnConvert which gave me an 8bit jpeg but I assume its a mistake in the metadata then.

Its milliseconds but I am converting millions of images 👍

**Comment by JimBobSquarePants** [comment]
@CoenraadS Actually you are correct! We should support 8 bit encoding!

**Comment by antonfirsov** [comment]
I'm not sure whether there are more performance-related bits which are not covered by the text in #808, #809 and #642.

If not, we can close this issue in favor of those.

**Comment by CoenraadS** [comment]
Turns out SkiaSharp doesn't support 8 bit resizing either, it seems not many libraries actually support 8 bit operations other then decoding.

**Comment by WrongBit** [comment]
Well... any progress, guys?
I also have similar task: grayscale image I want to save as JPEG/8-bit depth.
Maybe somebody already implement it? I feel it's not so hard task.

**Comment by JimBobSquarePants** [comment]
> Maybe somebody already implement it? I feel it's not so hard task.

PRs are welcome!

**Comment by antonfirsov** [comment]
Yeah, core team handles this with priority, but community PR-s are welcome.

The feature request is tracked under #808. All discussions should happen under that issue, please don't reply here.
